simul/build.go

```
@@ -311,29 +318,42 @@ func getStartStop(rcs int) (int, int) {
 
 // getRunWait returns either the command-line value or the value from the runconfig
 // file
-func getRunWait(rc *platform.RunConfig) time.Duration {
+func getRunWait(rc *platform.RunConfig) (time.Duration, error) {
 	rcWait, err := rc.GetDuration("runwait")
+	if err == platform.ErrorFieldNotPresent {
+		return runWait, nil
+	}
 	if err == nil {
-		return rcWait
+		return rcWait, nil
 	}
-	return runWait
+	return 0, err
 }
 
 // getExperimentWait returns
 // 1. the command-line value
 // 2. the value from runconfig
 // 3. #runconfigs * runWait
-func getExperimentWait(rcs []*platform.RunConfig) time.Duration {
+func getExperimentWait(rcs []*platform.RunConfig) (time.Duration, error) {
 	if experimentWait > 0 {
-		return experimentWait
+		return experimentWait, nil
 	}
 	rcExp, err := rcs[0].GetDuration("experimentwait")
 	if err == nil {
-		return rcExp
+		return rcExp, nil
+	}
+	// Probably a parse error parsing the duration.
+	if err != platform.ErrorFieldNotPresent {
+		return 0, err
 	}
+
+	// Otherwise calculate a useful default.
 	wait := 0 * time.Second
 	for _, rc := range rcs {
-		wait += getRunWait(rc)
+		w, err := getRunWait(rc)
+		if err != nil {
+			log.Fatal("RunWait:", err)
```

we can return an error here instand